core/src/test/java/org/elasticsearch/index/query/SpanContainingQueryBuilderTest.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.spans.SpanContainingQuery;
+import org.apache.lucene.search.spans.SpanQuery;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class SpanContainingQueryBuilderTest extends BaseQueryTestCase<SpanContainingQueryBuilder> {
+
+    @Override
+    protected Query doCreateExpectedQuery(SpanContainingQueryBuilder testQueryBuilder, QueryParseContext context) throws IOException {
+        SpanQuery big = (SpanQuery) testQueryBuilder.big().toQuery(context);
+        SpanQuery little = (SpanQuery) testQueryBuilder.little().toQuery(context);
+        return new SpanContainingQuery(big, little);
+    }
+
+    @Override
+    protected SpanContainingQueryBuilder doCreateTestQueryBuilder() {
+        SpanTermQueryBuilder bigQuery = new SpanTermQueryBuilderTest().createTestQueryBuilder();
+        // we need same field name and value type as bigQuery for little query
+        Object bigValue = bigQuery.value;
+        Object littleValue;
+        if (bigValue instanceof Boolean) {
```

Comment:
just my personal preference, I don't like instanceof checks that's it. you are free to leave them if you prefer them ;)